extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java

```diff
@@ -278,4 +278,179 @@ public void transform(TransformationContext ctx) {
+                    liveReload.setContextObject(CachedHealthUI.class, cached);
+                    Runtime.getRuntime().addShutdownHook(new Thread(cached, "Health UI Shutdown Hook"));
+                }
+                try {
+                    AppArtifact artifact = getHealthUiArtifact(curateOutcomeBuildItem);
+                    Path tempDir = Files.createTempDirectory(TEMP_DIR_PREFIX).toRealPath();
+                    extractHealthUi(artifact, tempDir);
+                    updateApiUrl(tempDir.resolve(FILE_TO_UPDATE), healthPath);
+                    cached.cachedDirectory = tempDir.toAbsolutePath().toString();
+                    cached.cachedHealthPath = healthPath;
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            }
+            Handler<RoutingContext> handler = recorder.uiHandler(cached.cachedDirectory,
+                    httpRootPath.adjustPath(health.ui.rootPath));
+            routeProducer.produce(new RouteBuildItem(health.ui.rootPath, handler));
+            routeProducer.produce(new RouteBuildItem(health.ui.rootPath + "/*", handler));
+            notFoundPageDisplayableEndpointProducer
+                    .produce(new NotFoundPageDisplayableEndpointBuildItem(health.ui.rootPath + "/"));
+        } else if (health.ui.alwaysInclude) {
+            AppArtifact artifact = getHealthUiArtifact(curateOutcomeBuildItem);
+            //we are including in a production artifact
+            //just stick the files in the generated output
+            //we could do this for dev mode as well but then we need to extract them every time
+            for (Path p : artifact.getPaths()) {
+                File artifactFile = p.toFile();
+                try (JarFile jarFile = new JarFile(artifactFile)) {
+                    Enumeration<JarEntry> entries = jarFile.entries();
+
+                    while (entries.hasMoreElements()) {
+                        JarEntry entry = entries.nextElement();
+                        if (entry.getName().startsWith(HEALTH_UI_WEBJAR_PREFIX) && !entry.isDirectory()) {
+                            try (InputStream inputStream = jarFile.getInputStream(entry)) {
+                                String filename = entry.getName().replace(HEALTH_UI_WEBJAR_PREFIX + "/", "");
+                                byte[] content = FileUtil.readFileContents(inputStream);
+                                if (entry.getName().endsWith(FILE_TO_UPDATE)) {
+                                    content = updateApiUrl(new String(content, StandardCharsets.UTF_8), healthPath)
+                                            .getBytes(StandardCharsets.UTF_8);
+                                }
+                                if (IGNORE_LIST.contains(filename)) {
+                                    ClassLoader classLoader = SmallRyeHealthProcessor.class.getClassLoader();
+                                    try (InputStream resourceAsStream = classLoader
```

Comment:
This may throw a NPE

Reply:
Hi @gastaldi - what might be null here ? The classLoader ?